cmd/ssh.go

```diff
@@ -176,47 +83,74 @@ func SSH(cmd *cobra.Command, args []string) error {
 		}
 	}
 
-	for {
-		select {
-		case e := <-execCh:
-			if e.Status == types.StatusRunning {
-				if e.Connection == nil {
-					ui.Errorf("❌ Something unexpected happened. No connection info found for session %q", e.ID)
-					ui.Infof("Run `unweave ls` to see the status of your session and try connecting manually.")
-					os.Exit(1)
-				}
-				ui.Infof("🚀 Session %q up and running", e.ID)
+	if withOpenVSCode {
+		return handleSSHConnWithOpenVSCode(ctx, execCh, isNew, sshArgsByKey, errCh)
+	} else {
+		return handleSSHConn(ctx, execCh, isNew, sshArgsByKey, errCh)
+	}
+}
 
-				if err := ssh.RemoveKnownHostsEntry(e.Connection.Host); err != nil {
-					// Log and continue anyway. Most likely the entry is not there.
-					ui.Debugf("Failed to remove known_hosts entry: %v", err)
-				}
+// parseSSHArgs dynamically parses the SSH connection arguments and returns them as a map.
+func parseSSHArgs(args []string) (map[string]string, error) {
+	sshArgs := make(map[string]string)
 
-				if err := ssh.AddHost("uw:"+e.ID, e.Connection.Host, e.Connection.User, e.Connection.Port); err != nil {
-					ui.Debugf("Failed to add host to ssh config: %v", err)
-				}
+	if len(args) < 1 {
+		return sshArgs, nil
+	}
+	if len(args) == 1 {
+		sshArgs["execRef"] = args[0]
+		return sshArgs, nil
+	}
+	if args[0] != "--" {
+		return nil, errors.New("invalid arguments format. Use '--' as the separator")
+	}
 
-				defer func() {
-					if e := ssh.RemoveHost("uw:" + e.ID); e != nil {
-						ui.Debugf("Failed to remove host from ssh config: %v", e)
-					}
-				}()
+	// Iterate over the arguments starting from the second one
+	for i := 1; i < len(args); i += 2 {
+		key := strings.TrimPrefix(args[i], "--")
+		if i+1 >= len(args) {
+			return nil, errors.New("invalid arguments format. Missing value for flag: " + key)
+		}
+		value := args[i+1]
+		sshArgs[key] = value
+	}
```

Comment:
This is the function that parses SSH arguments. It makes sure that if the first argument is some sort of non flag ID, then the ID gets used as the 'execRef'. Otherwise it looks for arguments formatted by the `--seperator`.

Reply:
I'm not sure this will work. This seems to assume that the all arguments to the ssh command are passed as flags and that all flag keys start with `--`. That is certainly not true. We don't want to mess with parsing the ssh args. We basically want to take everything after `--` as is and forward it to the ssh command